tests/test_configuration.py

```diff
@@ -67,11 +70,22 @@ def test_gets_value_from_env_with_json_parser(self) -> None:
 
         assert env_cfg.value == {"test": "ttt"}
 
+    @patch.dict(environ, {"RANDOM_VAR": "not-none"})
+    def test_reset_sets_value_to_none_forcing_fetching_it_once_again(self) -> None:
+        env_cfg = EnvValue[str]("RANDOM_VAR")
+        assert env_cfg.value == "not-none"
+
+        env_cfg.reset()
+
+        with patch.dict(environ, {"RANDOM_VAR": "still-not-none"}):
+            env_cfg = EnvValue[str]("RANDOM_VAR")
```

Comment:
You cannot create a completely new variable as this won't confirm the `.reset()` method logic.